build.gradle.kts

```diff
@@ -92,43 +83,38 @@ class ClearDependencies : ComponentMetadataRule {
     }
 }
 
-tasks {
-    compileKotlin {
-        kotlinOptions.jvmTarget = "11"
-    }
-    compileTestKotlin {
-        kotlinOptions.jvmTarget = "11"
-    }
+kotlin {
+    jvmToolchain(11)
```

Comment:
I wonder if we can upgrade to 17

Reply:
Looks like the build still succeeds; but we would probably also have to update the GitHub workflows to use JDK 17 as well, currently it specifies JDK 11. With the JVM toolchain of this PR Gradle should automatically download a matching JDK. But I hope we can avoid this (and therefore improve build speed) by directly using JDK 17 for the GitHub workflows.

Also, not sure if we need to adjust the server where Arisa is currently running to have JDK 17 installed. I am not familiar with the setup there at all.

Should we try switching to Java 17 in a separate PR?

Reply:
Server would have to be upgradef first indeed